src/cpp/lpnamer/input_reader/LpFilesExtractor.cpp

```diff
@@ -24,11 +51,20 @@ void LpFilesExtractor::ExtractFiles() const {
         << log_location << msg.str();
     throw ErrorWithAreaFile(msg.str(), log_location);
   }
-  std::filesystem::rename(vect_area_files[0],
-                          xpansion_output_dir_ / "area.txt");
 
-  const auto vect_interco_files =
-      archive_reader.ExtractPattern("interco*.txt", "", xpansion_output_dir_);
+  if (mode_ == Mode::ARCHIVE) {
+    std::filesystem::rename(vect_area_files[0],
+                            xpansion_output_dir_ / "area.txt");
+  } else {
+    try {
+      std::filesystem::copy(vect_area_files[0],
```

Comment:
Why a copy rather than just rename ?

Reply:
During implementation I had some issue with juste rename but it doesn't seem to cause issue now.